### PR TITLE
doc: add support status page - v1

### PR DESCRIPTION
**Pull request by jasonish**
Initial draft of support status page conversion:
https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Support_Status

Notes:
- Break out tiers into their own section as ReST table do not support headers
  within a table, so that context was lost.
- This needs review of content as well, which I have not done.
- Some stuff still to-do

Demo: https://jasonish-suricata.readthedocs.io/en/doc-support-status-v1/support-status.html

### CODE REVIEW

**Comment by jasonish** [comment]
Example using **bold** to provide an effect like a header row inside a table: https://jasonish-suricata.readthedocs.io/en/doc-support-status-v1/support-status.html#operation-modes

File: doc/userguide/support-status.rst [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+   +------------------+----------+----------+-------+--------------------------------+
+   |Distribution      |Version   |Support   |QA     |Notes                           |
+   +==================+==========+==========+=======+================================+
+   |CentOS            |Stream    |Core team |       |                                |
+   +------------------+----------+----------+-------+--------------------------------+
+   |Fedora            |Active    |Core team |       |                                |
+   +------------------+----------+----------+-------+--------------------------------+
+   |FreeBSD           |11        |Core team |       |                                |
+   +------------------+----------+----------+-------+--------------------------------+
+   |OpenBSD           |6.8       |Core team |       |                                |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:106]: 7.1/7.2 are current

File: doc/userguide/support-status.rst [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+   +------------------+----------+----------+-------+--------------------------------+
+   |OpenBSD           |6.9       |Core team |       |                                |
+   +------------------+----------+----------+-------+--------------------------------+
+   |OSX/macOS         |??        |Core team |       |                                |
+   +------------------+----------+----------+-------+--------------------------------+
+   |Windows/MinGW64   |Windows 10|Core team |       |5.0+                            |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:112]: think we can drop the `windows 10` and `5.0+`

File: doc/userguide/support-status.rst [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+~~~~~~~~~
+
+.. table::
+   :widths: 20 15 15 10 40
+   :width: 100%
+
+   +------------------+----------+----------+-------+--------------------------------+
+   |Architecture      |Version   |Support   |QA     | Notes                          |
+   +==================+==========+==========+=======+================================+
+   |Windows/MinGW32   |Windows 10|Core team |       |5.0+                            |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:125]: should we just drop this for 7?

File: doc/userguide/support-status.rst [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+.. table::
+   :widths: 15 15 30 40
+   :width: 100%
+
+   +-------------+-------------+---------------------------+---------------------------------------------+
+   |Architecture |Support      |QA                         |Notes                                        |
+   +=============+=============+===========================+=============================================+
+   |PPC64el      |             |Part of Fedora automated QA|Access can be arranged through IBM dev cloud |
+   +-------------+-------------+---------------------------+---------------------------------------------+
+   |PPC64        |Victor Julien|                           |No access except an old Mac G5 Victor has    |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:173]: can remove references to me here, that hw is no longer functional

should add riscv here

File: doc/userguide/support-status.rst [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+Tier 1
+^^^^^^
+
+.. table::
+   :width: 100%
+
+   +----------------+-------------------------+----+-----------------------------+
+   | Capture Type   | Maintainer              | QA | Notes                       |
+   +================+=========================+====+=============================+
+   |AF_PACKET       |Core team / Eric Leblond |    |Used by Security Onion, SELKS|
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:193]: maintainer should be core team only

File: doc/userguide/support-status.rst [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
+   +----------------+-------------------------+----+-----------------------------+
+   |NETMAP (FreeBSD)|Core team / Victor Julien|    |Used by OPNsense             |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:195]: maintainer just core team
used also by pfSense I believe

File: doc/userguide/support-status.rst [hunk 7]
```diff
@@ -0,0 +1,490 @@
...
+   |                |+ Aleksey Katargin       |    |                             |
+   +----------------+-------------------------+----+-----------------------------+
+   |NFQUEUE         |Core team / Andreas Herz |    |                             |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:198]: core team

File: doc/userguide/support-status.rst [hunk 8]
```diff
@@ -0,0 +1,490 @@
+Support Status
+==============
+
+Levels of Support
+-----------------
+
+Tier 1
+~~~~~~
+
+Core team develops and supports. Compiler errors or functional
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:10]: wonder if we should `s/Core team/OISF/g` for the entire doc

File: doc/userguide/support-status.rst [hunk 9]
```diff
@@ -0,0 +1,490 @@
...
+
+   +--------------------+-------------------------+----+---------------+
+   |Capture Type        |Maintainer               |QA  |Notes          |
+   +====================+=========================+====+===============+
+   |PF_RING             |Core team                |    |               |
+   +--------------------+-------------------------+----+---------------+
+   |AF_PACKET (eBPF/XDP)|Eric Leblond             |    |Bleeding edge  |
+   |                    |                         |    |               |
+   +--------------------+-------------------------+----+---------------+
+   |NETMAP (Linux)      |Core team / Victor Julien|    |               |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:217]: just core team / oisf

File: doc/userguide/support-status.rst [hunk 10]
```diff
@@ -0,0 +1,490 @@
...
+
+.. table::
+   :width: 100%
+
+   +--------------------+-------------------------+----+---------------+
+   |Capture Type        |Maintainer               |QA  |Notes          |
+   +====================+=========================+====+===============+
+   |PF_RING             |Core team                |    |               |
+   +--------------------+-------------------------+----+---------------+
+   |AF_PACKET (eBPF/XDP)|Eric Leblond             |    |Bleeding edge  |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:214]: should this move to community support?

File: doc/userguide/support-status.rst [hunk 11]
```diff
@@ -0,0 +1,490 @@
...
+   |                    |                         |    |               |
+   +--------------------+-------------------------+----+---------------+
+   |NETMAP (Linux)      |Core team / Victor Julien|    |               |
+   |                    |+ Aleksey Katargin       |    |               |
+   +--------------------+-------------------------+----+---------------+
+
+Community
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:221]: wonder if it makes sense to split this between vendor and community, where a vendor would be about supporting their own hw, like napatech or endace

File: doc/userguide/support-status.rst [hunk 12]
```diff
@@ -0,0 +1,490 @@
...
+   :width: 100%
+
+   +------------+---------------------+----+-----------------------+
+   |Type        |Maintainer           |QA  |Notes                  |
+   +============+=====================+====+=======================+
+   |EVE         |Core team            |    |                       |
+   +------------+---------------------+----+-----------------------+
+   |fast.log    |Core team            |    |                       |
+   +------------+---------------------+----+-----------------------+
+   |Unified2    |Core team / Jason Ish|    |scheduled for removal  |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:313]: removed

File: doc/userguide/support-status.rst [hunk 13]
```diff
@@ -0,0 +1,490 @@
...
+
+.. table::
+   :width: 100%
+
+   +------------+------------------------------------+----+-----------------------+
+   |Type        |Maintainer                          |QA  |Notes                  |
+   +============+====================================+====+=======================+
+   |http.log    |Core team                           |    |                       |
+   +------------+------------------------------------+----+-----------------------+
+   |dns.log     |Core team                           |    |scheduled for removal  |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:333]: removed

File: doc/userguide/support-status.rst [hunk 14]
```diff
@@ -0,0 +1,490 @@
...
+Community
+^^^^^^^^^
+
+.. table::
+   :width: 100%
+
+   +------------+----------------+----+----------------+
+   |Type        |Maintainer      |QA  |Notes           |
+   +============+================+====+================+
+   |prelude     |Thomas ANDREJAK |    |                |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:349]: removed

File: doc/userguide/support-status.rst [hunk 15]
```diff
@@ -0,0 +1,490 @@
...
+Scheduled for Removal
+^^^^^^^^^^^^^^^^^^^^^
+
+.. table::
+   :width: 100%
+
+   +------------+----------------+----+--------------------------+
+   |Type        |Maintainer      |QA  |Notes                     |
+   +============+================+====+==========================+
+   |files-json  |                |    |Superceded by eve.fileinfo|
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:361]: removed

File: doc/userguide/support-status.rst [hunk 16]
```diff
@@ -0,0 +1,490 @@
...
+Tier 2
+^^^^^^
+
+.. table::
+   :width: 100%
+
+   +------------+-------------------------+----+----------------------------------------------+
+   |Protocol    |Maintainer               |QA  |Notes                                         |
+   +============+=========================+====+==============================================+
+   |ftp         |Core team                |    |Currently only used for ‘ftpbounce’ keyword   |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:400]: move to tier 1, remove note

File: doc/userguide/support-status.rst [hunk 17]
```diff
@@ -0,0 +1,490 @@
...
+   +------------+-------------------------+----+----------------------------------------------+
+   |rust/nfs    |Core team / Victor Julien|    |Rust support currently experimental           |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:402]: move tier 1, drop rust references

File: doc/userguide/support-status.rst [hunk 18]
```diff
@@ -0,0 +1,490 @@
...
+   +------------+-------------------------+----+----------------------------------------------+
+   |rust/dns    |Core team / Jason Ish    |    |Rust support currently experimental           |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:404]: remove, dns already in tier 1

File: doc/userguide/support-status.rst [hunk 19]
```diff
@@ -0,0 +1,490 @@
...
+Community
+^^^^^^^^^
+
+.. table::
+   :width: 100%
+
+   +------------+-------------------+----+-------------------------+
+   |Protocol    |Maintainer         |QA  |Notes                    |
+   +============+===================+====+=========================+
+   |modbus      |David Diallo       |    |                         |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:416]: was done in rust using sawp, need to think where it fits

File: doc/userguide/support-status.rst [hunk 20]
```diff
@@ -0,0 +1,490 @@
...
+   +------------+-------------------+----+-------------------------+
+   |enip/cip    |Kevin Wong         |    |                         |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:418]: think Kevin is not involved anymore

File: doc/userguide/support-status.rst [hunk 21]
```diff
@@ -0,0 +1,490 @@
...
+   +------------+-------------------+----+-------------------------+
+   |rust/ntp    |Pierre Chifflier   |    |Experimental Rust support|
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:420]: at least tier 2? rust no longer experimental/optional

File: doc/userguide/support-status.rst [hunk 22]
```diff
@@ -0,0 +1,490 @@
...
+^^^^^^^^^^^^
+
+.. table::
+   :widths: 30 25 20 25
+   :width: 100%
+
+   +------------+----------+-----+-----+
+   |Method      |Maintainer|QA   |Notes|
+   +============+==========+=====+=====+
+   |pcie (tile) |          |     |     |
```
Comment (victorjulien) [inline doc/userguide/support-status.rst:489]: has been removed

File: doc/userguide/support-status.rst (hunk 8 above)
Comment (norg) [reply doc/userguide/support-status.rst:10]: Since we don't have two different teams within OISF I would agree or at least add OISF so either `OISF team` or `OISF core team`

File: doc/userguide/support-status.rst (hunk 10 above)
Comment (norg) [reply doc/userguide/support-status.rst:214]: With more and more people trying to use it, it might be worth to make it obvious that it's not 100% supported by us or we will have to invest more to bump it to that kind of Tier.

File: doc/userguide/support-status.rst (hunk 11 above)
Comment (norg) [reply doc/userguide/support-status.rst:221]: I fully agree, it makes it also more clear that there are different parts of the community

File: doc/userguide/support-status.rst [hunk 23]
```diff
@@ -0,0 +1,490 @@
...
+Core team develops and supports, sometimes with help from community
+members. Compiler errors or functional failures block git merges.
+Functionality problems such as ‘known issues’ or might still going
+into releases. Functionality may be disabled by default.
+
+Community
+~~~~~~~~~
+
+When a feature of Suricata is community supported, it means the
+OISF/Suricata development team won’t support it. This to avoid
```
Comment (norg) [inline doc/userguide/support-status.rst:26]: I would rephrase it do `won't directly support it` since we still might support it a bit

File: doc/userguide/support-status.rst (hunk 7 above)
Comment (norg) [reply doc/userguide/support-status.rst:198]: we should remove all personal names maybe?

**Comment by jasonish** [comment]
Overall, hows the format?

**Comment by victorjulien** [comment]
> Overall, hows the format?

Great.

File: doc/userguide/support-status.rst (hunk 8 above)
Comment (jufajardini) [reply doc/userguide/support-status.rst:10]: I think I would go with OISF team, as `OISF core team` might be interpreted as "there is a non-core OISF team"...

File: doc/userguide/support-status.rst [hunk 24]
```diff
@@ -0,0 +1,490 @@
...
+
+* submitter must commit to maintaining it:
+
+  - make sure code compiles and correctly functions after Suricata
+    and/or external (e.g. library) changes.
+  - support users when they encounter problems on the mailinglists and
+    redmine tickets
+
+* the code will be disabled by default and will not become part of the
+  QA setup. This means it will be enabled only by a —enable- configure
```
Comment (jufajardini) [inline doc/userguide/support-status.rst:40]: nit: format `--enable-`  as inline code, so that the two dashes won't be changed to some other symbol.

File: doc/userguide/support-status.rst [hunk 25]
```diff
@@ -0,0 +1,490 @@
...
+maintaining it and no-one steps up to take over.
+
+Unmaintained
+~~~~~~~~~~~~
+
+When a feature is unmaintained it is very likely broken and may be
+(partially) removed during cleanups and code refactoring. No end-user
+support is done by the core team. If someone wants to help maintain
+and support such a feature, we recommend talking to the core team
+before spending a lot of time on it.
```
Comment (jufajardini) [inline doc/userguide/support-status.rst:56]: May be good to add a reference to: https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html#communication-is-key as we mention such things here and the channels for folks to reach out.

File: doc/userguide/support-status.rst [hunk 26]
```diff
@@ -0,0 +1,490 @@
...
+  - make sure code compiles and correctly functions after Suricata
+    and/or external (e.g. library) changes.
+  - support users when they encounter problems on the mailinglists and
+    redmine tickets
+
+* the code will be disabled by default and will not become part of the
+  QA setup. This means it will be enabled only by a —enable- configure
+  flag.
+
+If the feature get lots of traction, and/or if the team just considers
```
Comment (jufajardini) [inline doc/userguide/support-status.rst:43]: nit: s/get/gets

File: doc/userguide/support-status.rst (hunk 3 above)
Comment (jasonish) [reply doc/userguide/support-status.rst:125]: Yes, I think we should.

File: doc/userguide/support-status.rst (hunk 8 above)
Comment (jasonish) [reply doc/userguide/support-status.rst:10]: Trying simply `OISF`.

File: doc/userguide/support-status.rst (hunk 10 above)
Comment (jasonish) [reply doc/userguide/support-status.rst:214]: Moved to community.

File: doc/userguide/support-status.rst (hunk 19 above)
Comment (jasonish) [reply doc/userguide/support-status.rst:416]: Left as community, no maintainer.

File: doc/userguide/support-status.rst (hunk 11 above)
Comment (jasonish) [reply doc/userguide/support-status.rst:218]: I've added a *Vendor* support tier, documented it and moved Napatech into a *Vendor* section.

**Comment by jasonish** [comment]
Latest changes can be seen at: https://jasonish-suricata.readthedocs.io/en/doc-support-status-v1/support-status.html

**Comment by victorjulien** [comment]
Now I wonder if we should:
- have a "packagers guide" to make distro packagers aware of the doc
- match our defaults to this list: tier 1 (enabled by default), tier 2 (disabled by default), rest (compile time option not compiled in by default)

**Comment by jufajardini** [comment]
> Now I wonder if we should:
> 
> * have a "packagers guide" to make distro packagers aware of the doc

+1

**Comment by jasonish** [comment]
> * match our defaults to this list: tier 1 (enabled by default), tier 2 (disabled by default), rest (compile time option not compiled in by default)

How do you see this working for something like `PF_RING` or `Netmap`? They're tier 2 and I believe a compile time option.  Should they instead be built if the libraries exist even if not explicit `--enable` flag is given?  Given they're capture methods, they would still be disabled by default.

I guess this would also put `enip` behind a compile time flag? Make `modbus` compiled and disabled by default.  DNP3 is tier 1, yet disabled by default as well.  I'm not so sure this list maps directly what we enable by default.

**Comment by jasonish** [comment]
> * have a "packagers guide" to make distro packagers aware of the doc

Yes, I have some ideas here so will draft something.

**Comment by jasonish** [comment]
Feedback integrated, new PR: https://github.com/OISF/suricata/pull/9163
